drivers/console/ram_console.c

```diff
@@ -11,23 +12,36 @@
 #include <zephyr/sys/printk.h>
 #include <zephyr/device.h>
 #include <zephyr/init.h>
+#include <zephyr/linker/devicetree_regions.h>
 
 extern void __printk_hook_install(int (*fn)(int));
 extern void __stdout_hook_install(int (*fn)(int));
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(ram_console), okay)
```

Comment:
what about using a `chosen` node instead?

Reply:
Will update. Thanks a lot for your comments!

Reply:
updated, please help to review again. Thanks!